jekyll/assets/js/user.js

```diff
@@ -0,0 +1,76 @@
+function setUserData(userData) {
+  window.userData = userData;
+  // emit an event to let the system know that userData is ready/has changed
+  const userDataReady = new CustomEvent("userDataReady");
```

Comment:
this is good but since there's no event emitted once optimizely is ready, how do we trigger experiment code?

Reply:
There is: https://github.com/circleci/circleci-docs/blob/ae050127cb247c91388fce0bfc1289da561c45cb/src-js/optimizelyClient.js#L26 

You have a good point because the current implementation of `getVariationName` will not work if we want to **wait** until we know what to display `control`/`treatment`. The current implementation works really well for additive experiments (control has nothing and treatment has something new) but in the case you will want more than that, then we need to prevent the `control`/`treatment` flashing. 

I'll work on this today!

Reply:
@nramabad: Addressed your concern and added a lot of comments to make sure the logic is clear to everyone: https://github.com/circleci/circleci-docs/pull/5630/commits/cbe5892aa23d14f8f26e12e0547fde1a5a4574e4